### PR TITLE
Update repo-infra dependencies

### PR DESCRIPTION
**Pull request by fejta**
* bazel-gazelle 0.20.0
* golangci-lint 1.23.7
* bazel-toolchains 2.1.1 -- supports bazel 2.1.1 oob
* protobuf 3.11.4
* rules_go 0.22.1
* Load rules_proto, rules_nodejs and buildtools in load.bzl

/assign @BenTheElder

### CODE REVIEW

**Comment by fejta** [comment]
Passing tests now (needed the golangci lint needs the golang sources, which are now passed into this test target)

File: .bazelrc [hunk 1]
```diff
@@ -26,6 +26,9 @@ build:remote --remote_timeout=3600
 # Alt: --google_credentials=some_file.json
 build:remote --google_default_credentials=true
 
+# Improve cache hit rate
+build:remote --incompatible_strict_action_env=true
```
Comment (BenTheElder) [inline .bazelrc]: huh yep, this WAS on by default at one point https://github.com/bazelbuild/bazel/issues/7026 https://github.com/bazelbuild/bazel/issues/6648

**Pull request merged**